src/test/kotlin/com/igorwojda/integer/factorial/challenge.kt

```diff
@@ -4,7 +4,7 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun factorial(n: Int): Int {
-    TODO("not implemented")
+    return if (n == 0) 1 else (1..n).toList().reduce { acc: Int, current: Int -> acc * current }
```

Comment:
Nice use of folding operations. Please also write a recursive implementation and compare the running time of the two solutions.
If you're curious, look at tail recursive functions (https://kotlinlang.org/docs/reference/functions.html#tail-recursive-functions). See if you can write factorial as a tailrec function.